src/ClientHandle.h

```diff
@@ -453,7 +453,12 @@ class cClientHandle  // tolua_export
 		// TODO: Add Kicking here as well
 	} ;
 	
+#ifdef ANDROID
+#pragma message "Android <atomic> support is incomplete or broken - using plain variable (subject to data races!)"
+	eState m_State;
```

Comment:
As I said, if thread safety is less important than performance, you should make cCriticalSection a noop. A minimal std::atomic implantation using cCriticalSection would not be much work. It doesn't need to be fully standard complaint, just provide the interfaces we use.